include/inexor/vulkan-renderer/shader.hpp

```diff
@@ -1,26 +1,62 @@
 #pragma once
 
-#include "inexor/vulkan-renderer/tools/file.hpp"
-
 #include <vulkan/vulkan.h>
 
 #include <string>
+#include <vector>
 
 namespace inexor::vulkan_renderer {
 
-class Shader : public tools::File {
-public:
-    Shader() = default;
-
-    ~Shader() = default;
-
+class Shader {
+private:
+    VkDevice device;
+    VkShaderStageFlagBits type;
     std::string name;
+    std::string entry_point;
 
-    std::string entry_name;
-
-    VkShaderStageFlagBits type;
+    VkShaderModule shader_module;
 
-    VkShaderModule module;
+public:
+    /// @brief Creates a shader from memory.
+    /// @param device [in] The Vulkan device which will be used to create the shader module.
+    /// @param type [in] The shader type (vertex shader, fragment shader, tesselation shader..).
+    /// @param name [in] The internal name of the shader module.
+    /// @param code [in] The SPIR-V shader code.
+    /// @param entry_point [in] The entry point of the shader code, in most cases just "main".
+    Shader(VkDevice device, VkShaderStageFlagBits type, const std::string &name, const std::vector<char> &code, const std::string &entry_point = "main");
+
+    /// @brief Creates a shader from a SPIR-V file.
+    /// @param device [in] The Vulkan device which will be used to create the shader module.
+    /// @param type [in] The shader type (vertex shader, fragment shader, tesselation shader..).
+    /// @param name [in] The internal name of the shader module.
+    /// @param file_name [in] The name of the SPIR-V shader file.
+    /// @param entry_point [in] The entry point of the shader code, in most cases just "main".
+    Shader(VkDevice device, VkShaderStageFlagBits type, const std::string &name, const std::string &file_name, const std::string &entry_point = "main");
+
+    /// Delete the copy constructor so Shaders are move-only objects.
+    Shader(const Shader &) = delete;
```

Comment:
Can you place the "default"/simple constructors at the top?

Reply:
Same for cpp then.

Reply:
sure.